text/0134-remove-span-status.md

```diff
@@ -0,0 +1,88 @@
+
+## Motivation
+
+During numerous heated discussions in various pull requests (see
+<https://github.com/open-telemetry/opentelemetry-specification/issues/599>
+<https://github.com/open-telemetry/opentelemetry-specification/pull/697>
+<https://github.com/open-telemetry/opentelemetry-specification/pull/427>
+<https://github.com/open-telemetry/oteps/pull/69>
+<https://github.com/open-telemetry/oteps/pull/86>
+<https://github.com/open-telemetry/oteps/pull/123>)
+and during Errors Working Group meetings a lot of concerns were raised about suitability of the current `Span.Status`
+field encoded as Google RPC code for recording a result of arbitrary monitored operations from various domains
+(http requests, database queries, file system operations etc). It is hard to define clear and umambigous mapping
+from those domain to a fixed enum originated from RPC.
+
+Therefore the proposal is to remove the `Span.Status` field from Trace API and OTLP.
+
+## Internal details
+
+Several areas are affected by this change and will require the coordinate effort to make this happen.
+
+### Trace API Specification
+
+* Remove all mentions of `Status` and `Set Status` API from Span section.
+* Remove whole `Status` section.
+
+### Trace SDK Specification
+
+No changes needed.
+
+### Trace SDK exporters
+
+Remove translation rules pertaining to `Span.Status`.
+
+### Span semantic conventions
+
+* From HTTP conventions, remove `Status` section.
+* In gRPC conventions, replace the requirement to set span status with the requirement to set span attribute `grpc.status`
+to the canonical status code of the gRPC response.
+
+### OTLP data definition
+
+Rename `Span.status` field to `Span.deprecated_status` field.
+
+### Language API and SDK
+
+* Every language must remove status related code from their implementations of Trace API and SDK.
+* Every language must remove all code in their span exporters which serializes span status.
+* All instrumentation libraries, both manual and automated, must remove all calls to the removed `Span.setStatus` API.
+* All instrumentation libraries, both manual and automated, must adapt new semantic conventions for gRPC spans.
+
+### Collector
+
+Collector will have a transition period during which it has a special handling both for old data,
+which has `Span.Status` present, and new data without Status.
+This will allow for older pipelines, still relying on `Span.Status` to continue to work with as little disruption as possible.
+The exact duration of this transition period will be left for Collector's maintainers to decide.
+
+* If incoming OTLP data has `Span.Status` present, that will NOT be removed and will be passed on as-is.
+* If incoming OTLP data has `Span.Status` present, Collector will translate that to temporary semantic attributes as follows:
+
+|Status|Tag Key| Tag Value|
+|--|--|--|
+|StatusCanonicalCode | `otel.deprecated_status_code` | Name of the code, if that differs from ‘OK’ |
```

Comment:
What value do we set if it is 'OK'? Do we omit the `otel.deprecated_status_code` attribute?

Reply:
Yes, that was my idea. And probably should skip setting `otel.deprecated_status_message` as well

Reply:
Sounds good, please make it explicit in the text.